packages/sdk/src/assets/nft.ts

```diff
@@ -0,0 +1,126 @@
+import {numberToHex} from 'viem'
+
+import {NotMintedNftError} from '../errors/assets/not-minted-nft.error'
+import {IsNotMinted} from '../utils/decorators'
+import {HolographNFTMetadata, validate} from './nft.validation'
+
+export class NFT {
+  isMinted: boolean
+  private metadata: HolographNFTMetadata
+  private owner?: string
+  private chainId?: string | number
+  private file?: string
+  private tokenId?: string // Decimal tokenId string
+  private collectionAddress?: string
+  private mintTx?: string
+
+  constructor(metadata: HolographNFTMetadata) {
+    this.metadata = metadata
+    this.isMinted = false
+  }
+
+  getMetadata() {
+    return this.metadata
+  }
+
+  getName() {
+    return this.metadata.name
+  }
+
+  getDescription() {
+    return this.metadata.description
+  }
+
+  getCreator() {
+    return this.metadata.creator
+  }
+
+  // AKA the NFT traits
+  getAttributes() {
+    return this.metadata.attributes
+  }
+
+  getOwner() {
+    return this.owner
+  }
+
+  getFile() {
+    return this.file
+  }
+
+  getTokenId() {
+    if (!this.chainId || !this.tokenId) throw new NotMintedNftError(this.getTokenId.name)
```

Comment:
I believe that it's better to keep the same pattern for this type of verification. This was a suggestion made by @alexanderattar in the CLI.

```ts
if(this.chainId !== undefined || this.token !== undefined) {
   throw new NotMintedNftError(this.getTokenId.name)
}
```

Reply:
I think this is not what I was expecting. I wanna throw an error either if `this.chainId` or `this.tokenId` got any `falsy` value, not only `undefined`